xarray/core/datatree.py

```diff
@@ -1141,11 +1152,10 @@ def depth(item) -> int:
             # Sort keys by depth so as to insert nodes from root first (see GH issue #9276)
             for path, data in sorted(d_cast.items(), key=depth):
                 # Create and set new node
-                node_name = NodePath(path).name
                 if isinstance(data, DataTree):
                     new_node = data.copy()
                 elif isinstance(data, Dataset) or data is None:
-                    new_node = cls(name=node_name, dataset=data)
```

Comment:
We don't need to set `name` explicitly because it is already guaranteed to be consistent (via `NamedNode._post_attach`).